pkg/service/service.go

```diff
@@ -136,101 +136,102 @@ func (s *DeviceService) Stop(force bool) {
 
 // selfRegister register device service itself onto metadata.
 func (s *DeviceService) selfRegister() error {
-	s.LoggingClient.Debug("Trying to find Device Service: " + s.ServiceName)
+	addr, err := s.createAndUpdateAddressable()
+	if err != nil {
+		s.LoggingClient.Error(fmt.Sprintf("createAndUpdateAddressable failed: %v", err))
+		return err
+	}
+
+	newDeviceService := contract.DeviceService{
+		Name:           s.ServiceName,
+		Labels:         s.config.Service.Labels,
+		OperatingState: contract.Enabled,
+		Addressable:    *addr,
+		AdminState:     contract.Unlocked,
+	}
+	newDeviceService.Origin = time.Now().UnixNano() / int64(time.Millisecond)
 
 	ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.New().String())
+	s.LoggingClient.Debug("Trying to find DeviceService: " + s.ServiceName)
 	ds, err := s.edgexClients.DeviceServiceClient.DeviceServiceForName(ctx, s.ServiceName)
-
 	if err != nil {
 		if errsc, ok := err.(types.ErrServiceClient); ok && (errsc.StatusCode == http.StatusNotFound) {
-			s.LoggingClient.Info(fmt.Sprintf("Device Service %s doesn't exist, creating a new one", s.ServiceName))
-			ds, err = s.createNewDeviceService()
+			s.LoggingClient.Info(fmt.Sprintf("DeviceService %s doesn't exist, creating a new one", s.ServiceName))
+			id, err := s.edgexClients.DeviceServiceClient.Add(ctx, &newDeviceService)
+			if err != nil {
+				s.LoggingClient.Error(fmt.Sprintf("Failed to add Deviceservice %s: %v", s.ServiceName, err))
+				return err
+			}
+			if err = common.VerifyIdFormat(id, "Device Service"); err != nil {
+				return err
+			}
+			// NOTE - this differs from Addressable and Device Resources,
+			// neither of which require the '.Service'prefix
+			newDeviceService.Id = id
+			s.LoggingClient.Debug("New DeviceService Id: " + newDeviceService.Id)
 		} else {
 			s.LoggingClient.Error(fmt.Sprintf("DeviceServicForName failed: %v", err))
 			return err
 		}
 	} else {
-		s.LoggingClient.Info(fmt.Sprintf("Device Service %s exists", ds.Name))
+		s.LoggingClient.Info(fmt.Sprintf("Updating DeviceService %s", ds.Name))
```

Comment:
change the message to
"DeviceService %s exists, updating it"

Reply:
updated.